client/src/app/notifications/Notification.js

```diff
@@ -69,6 +70,29 @@ export default class Notification extends PureComponent {
     this.props.close();
   }
 
+  getContent() {
+    const {
+      content,
+      close
+    } = this.props;
+
+    // allow button to close notification
+    if (content && content.type === 'button') {
+      const contentWithButton = assign({},content);
+
+      const onClick = {
+        onClick:  ()=> {
+          content.props.onClick(); close();
```

Comment:
This still looks like we are monkey patching here. I understand why we are doing it (lazy adding props), but maybe we can rely on React API instead, something like

```js
 const onClick = ()=> {
  content.props.onClick(); close();
};

return React.cloneElement(
  content,
  { onClick }
);
```

Reply:
Smart thinking, makes much more sense. Thanks!